scripts/generator.py

```diff
@@ -40,24 +40,36 @@ def main():
         intermediate_fields = ecs_helpers.fields_subset(subset, intermediate_fields)
 
     (nested, flat) = schema_reader.generate_nested_flat(intermediate_fields)
-    intermediate_files.generate(nested, flat)
+
+    # default location to save files
+    out_dir = 'generated'
+    docs_dir = 'docs'
+    if args.out:
+        out_dir = os.path.join(args.out, out_dir)
+        docs_dir = os.path.join(args.out, docs_dir)
+
+    ecs_helpers.make_dirs(out_dir)
+    ecs_helpers.make_dirs(docs_dir)
+
+    intermediate_files.generate(nested, flat, out_dir)
     if args.intermediate_only:
         exit()
 
-    csv_generator.generate(flat, ecs_version)
-    es_template.generate(flat, ecs_version)
-    beats.generate(nested, ecs_version)
-    asciidoc_fields.generate(nested, flat, ecs_version)
+    csv_generator.generate(flat, ecs_version, out_dir)
+    es_template.generate(flat, ecs_version, out_dir)
+    beats.generate(nested, ecs_version, out_dir)
+    asciidoc_fields.generate(nested, flat, ecs_version, docs_dir)
 
 
 def argument_parser():
     parser = argparse.ArgumentParser()
     parser.add_argument('--intermediate-only', action='store_true',
                         help='generate intermediary files only')
-    parser.add_argument('--include', action='store',
+    parser.add_argument('--include', action='append',
```

Comment:
With the change for `append` here, if you pass in multiple `--include` flags it will only use the last one. This causes the values to be added to an array.

Reply:
Ah good point, didn't think of that. I don't think there should be any issue with switching it over to  `nargs`. I'll update it.